dyninstAPI/src/inst-aarch64.C

```diff
@@ -1389,37 +1422,257 @@ bool EmitterAARCH64::emitCallInstruction(codeGen &gen, func_instance *callee, bo
 
 void EmitterAARCH64::emitLoadShared(opCode op, Register dest, const image_variable *var, bool is_local, int size,
                                     codeGen &gen, Address offset) {
-    assert(0); //Not implemented
+    //assert(0); //Not implemented
+    //return;
+    
+    // create or retrieve jump slot
+    Address addr;
+    int stackSize = 0;
+
+    if(var == NULL) {
+        addr = offset;
+    }
+    else if(!is_local) {
+        addr = getInterModuleVarAddr(var, gen);
+    }
+    else {
+        addr = (Address)var->getOffset();
+    }
+
+    // load register with address from jump slot
+
+    printf("emitLoadShared addr 0x%lx curr adress 0x%lx offset %ld 0x%lx size %d\n", 
+            addr, gen.currAddr(), addr - gen.currAddr()+4, addr - gen.currAddr()+4, size);
+    Register scratchReg = gen.rs()->getScratchRegister(gen, true);
+
+    if (scratchReg == REG_NULL) {
+        assert(0);
+    }
+
+    emitMovePCToReg(scratchReg, gen);
+    Address varOffset = addr - gen.currAddr()+4;
```

Comment:
Just double-checking, is the associativity correct here?

Reply:
Yes! And I've properly spaced the expression.